pkg/agent/openflow/pipeline.go

```diff
@@ -604,14 +619,27 @@ func (f *featurePodConnectivity) tunnelClassifierFlow(tunnelOFPort uint32) bindi
 		Done()
 }
 
-// gatewayClassifierFlow generates the flow to mark the packets from the Antrea gateway port.
-func (f *featurePodConnectivity) gatewayClassifierFlow() binding.Flow {
-	return ClassifierTable.ofTable.BuildFlow(priorityNormal).
+// gatewayClassifierFlows generates the flow to mark the packets from the Antrea gateway port.
+func (f *featurePodConnectivity) gatewayClassifierFlows() []binding.Flow {
+	var flows []binding.Flow
+	for ipProtocol, gatewayIP := range f.gatewayIPs {
+		flows = append(flows, ClassifierTable.ofTable.BuildFlow(priorityHigh).
+			Cookie(f.cookieAllocator.Request(f.category).Raw()).
+			MatchInPort(f.gatewayPort).
+			MatchProtocol(ipProtocol).
+			MatchSrcIP(gatewayIP).
+			Action().LoadRegMark(FromGatewayRegMark).
+			Action().GotoStage(stageValidation).
+			Done())
+	}
+	// If the packet is from gateway but its source IP is not the gateway IP, it's considered external sourced traffic.
```

Comment:
just double-checking that the change to this function does not have an impact on any other feature (e.g. NetworkPolicy)?

Reply:
No, host-originated traffic will be masqueraded when forwarded to OVS. 
Even they are not masqueraded, those IPs can work in DSR mode, the backend Node doesn't know they are reachable via tunnel.